rust/vectordb/src/data/inspect.rs

```diff
@@ -0,0 +1,160 @@
+// Copyright 2023 Lance Developers.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::HashMap;
+
+use arrow::compute::kernels::{aggregate::bool_and, length::length};
+use arrow_array::{cast::AsArray, types::Int32Type, Array, RecordBatchReader};
+use arrow_ord::comparison::eq_dyn_scalar;
+use arrow_schema::DataType;
+
+use crate::error::{Error, Result};
+
+pub(crate) fn infer_dimension(arr: &dyn Array) -> Result<Option<i32>> {
+    if !matches!(arr.data_type(), DataType::List(_)) {
+        return Ok(None);
+    };
+    let list_arr = arr.as_list_opt::<i32>().expect("must be a list now");
```

Comment:
nit: iiuc, this `i32` really just means 32-bit wide, doesn't mean the data is actually integer. Could we alias this to be more clear?

```
type BIT_WIDTH_16 = i16;
...
```

Reply:
i32 means it is i32 as offset (which is `List` not `LargeList`).

Reply:
Ah ok that makes sense